Tests/BuildTests/BuildSystemDelegateTests.swift

```diff
@@ -0,0 +1,37 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SPMTestSupport
+import XCTest
+
+import var TSCBasic.localFileSystem
+
+final class BuildSystemDelegateTests: XCTestCase {
+    func testDoNotFilterLinkerDiagnostics() throws {
+        try fixture(name: "Miscellaneous/DoNotFilterLinkerDiagnostics") { fixturePath in
+            let (fullLog, _) = try executeSwiftBuild(fixturePath)
+            XCTAssertTrue(fullLog.contains("ld: warning: ignoring duplicate libraries: '-lz'"), "log didn't contain expected linker diagnostics")
```

Comment:
I think I'll probably need to make this macOS only, since these diagnostics may be different or not emitted at all by other linkers.

Reply:
Ah we also don't have `libz` on Linux :)

```
error: link command failed with exit code 1 (use -v to see invocation)
    /usr/bin/ld.gold: error: cannot find -lz
    /usr/bin/ld.gold: error: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    error: fatalError
```